crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -317,6 +321,9 @@ pub(crate) struct UseDefMap<'db> {
     /// [`SymbolState`] visible at end of scope for each symbol.
     public_symbols: IndexVec<ScopedSymbolId, SymbolState>,
 
+    /// [`SymbolState`] for each instance attribute.
+    instance_attribute_states: IndexVec<ScopedSymbolId, SymbolState>,
```

Comment:
nit: we could just call this `instance_attributes`.

We don't call `public_symbols` `public_symbol_states`, even though it is also an indexvec of `SymbolState`.